src/app/ReadClient.cpp

```diff
@@ -1271,6 +1272,15 @@ CHIP_ERROR ReadClient::DefaultResubscribePolicy(CHIP_ERROR aTerminationCause)
         return CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT;
     }
 
+    if (aTerminationCause.IsIMStatus() && ClusterStatusCode(aTerminationCause).GetStatus() == Status::InvalidAction)
+    {
+        // InvalidAction in most cases means some Interaction Model mismatch with the target device
```

Comment:
the `in most cases` seems vague, so does `some Interaction Model mismatch`. This feels like a "we do not know what the problem is, but this kind of works" ... we need stronger root cause investigation I believe.

Looking through the code on ReadHandler we seem to return InvalidAction on:
   - OnMessageReceived when not having received a StatusResponse
   - OnInitialRequest when processing the read or subscribe fails without setting a different IM status.

Trying to see what the specification states, it seems to say in `8.2.5.3 Incoming Action` that `If this layer receives a message for an action that is not expected semantically, has invalid action information, or has an error not described in this specification, a Status Response action with an INVALID_ACTION Status Code SHALL be generated as defined in Status Response Action, and the associated transaction and interaction SHALL terminate.`

And `Invalid Action` says `The action is malformed, has missing fields, or fields with invalid values. Action not carried out.`

Maybe the the invalid action definition would help here, however I would like a sanity review from someone more familiar with the spec and how this works. @bzbarsky-apple @tennessee-google @yunhanw-google please help with opinions here. I sugges:

```suggestion
        // InvalidAction means "The action is malformed, has missing fields, or fields with invalid 
        // values. Action not carried out." according to the spec and retrying will not fix this as we would
        // send out the same parameters (that are malformed, missing fields or invalid) so there is
        // no point to retry.
```